src/actions/playerActions.js

```diff
@@ -44,58 +44,52 @@ export const getDevices = () => dispatch => {
     .catch(e => console.log(e));
 };
 
-export const getCurrentPlaying = () => dispatch => {
-  playerService
-    .getCurrentlyPlaying()
-    .then(track => {
-      dispatch({ type: GET_CURRENT_PLAYING, payload: track });
-    })
-    .catch(e => console.log(e));
+export const getCurrentlyPlaying = () => async dispatch => {
+  const track = await playerService.getCurrentlyPlaying();
+  dispatch({ type: GET_CURRENT_PLAYING, payload: track });
+  // .then(track => {
+  //
+  // })
+  // .catch(e => console.log(e));
 };
 
-export const nextSong = () => dispatch => {
-  playerService
-    .next()
-    .then(_ => {
-      dispatch({ type: NEXT_SONG });
-    })
-    .catch(e => console.log(e));
+export const nextSong = () => async dispatch => {
+  await playerService.next();
+  dispatch({ type: NEXT_SONG });
+  setTimeout(() => dispatch(getCurrentlyPlaying()), 400);
```

Comment:
I wonder if it's better to include the delay inside the `getCurrentlyPlaying` function. Seems that you will need to wrap all the calls to this function. What o you think @linx05

Reply:
When the component load for the first time, I called getCurrentlyPlaying and It doesn't need the delay, that's why I didn't include the delay inside it, just did it on the actions that needed it, but it wouldn't be a huge deal if I move inside it!